find_posts.py

```diff
@@ -143,33 +190,54 @@ def get_user_posts(user, know_followings, server):
                 return all_posts
             
         except Exception as ex:
-            log(f"Error getting user posts for user {parsed_url[1]}: {ex}")
+            log(f"Error getting user posts for user {userName}: {ex}")
         return None
-    
+
+def get_user_posts_misskey(userName, webserver):
+    # query user info via search api
+    # we could filter by host but there's no way to limit that to just the main host on firefish currently
+    # on misskey it works if you supply '.' as the host but firefish does not
+    userId = None
     try:
-        user_id = get_user_id(parsed_url[0], parsed_url[1])
+        url = f'https://{webserver}/api/users/search-by-username-and-host'
+        resp = post(url, { 'username': userName })
+
+        if resp.status_code == 200:
+            res = resp.json()
+            for user in res:
+                if user['host'] is None:
+                    userId = user['id']
+                    break
+        else:
+            log(f"Error finding user {userName} from {webserver}. Status Code: {resp.status_code}")
+            return None
     except Exception as ex:
-        log(f"Error getting user ID for user {user['acct']}: {ex}")
+        log(f"Error finding user {userName} from {webserver}. Exception: {ex}")
```

Comment:
Does miskey have rate limits? If so, we should implement a clause for `429` responses